### ISSUE TITLE
AutoDMG v1.10b587 fails if any pkg a added on macOS Catalina 10.15 19A536g

### ISSUE THREAD

**Issue by vasquesbc** [MagerValp/AutoDMG#238]
## Description of issue

after a while the build fails with:

```
Workflow failed: Build failed (Installation of 'Package.pkg' failed with return code 1. See the log for details.)
Task exited with status 102
```

I thought that maybe my Packages are incompatible with Catalina - so i stripped them down step by step.
I ended up using an empty „Test Package.pkg“ without any scripts or payload - built with Stéphane Sudre's Packages 1.2.6 - it fails as well.
I did not find a clue inside the Logs - did i miss something?

## Expected behavior

AutoDMG should build a DMG, consisting of macOS Catalina & any additional packages

## Actual behavior

see above

## Steps to reproduce

1. Start AutoDMG v1.10b587 on macOS Catalina 10.15 19A536g
2. Drag the macOS Catalina 10.15 19A536g Installer App onto the AutoDMG-window
3. Drag any .pkg to the „Additional software“ table
4. Click „Build“
5. Wait a while

## Log file

[AutoDMG-2019-08-23.log](https://github.com/MagerValp/AutoDMG/files/3534161/AutoDMG-2019-08-23.log)

**Comment by jelockwood** [comment]
I tried a couple of installers -

1. A createuser pkg created using the latest copy of Greg Neagle's pycreateuserpkg script
2. A QuickAdd package generated from Jamf

The createuser pkg is something that historically was one of the very few installers to work with AutoDMG. With this I get no error during the building of the dmg but when it runs it clearly has not successfully run that installer pkg because after deploying the image it boots to the standard Apple Setup Wizard and this indicates there are no user accounts present.

With the Jamf QuickAdd the AutoDMG build process fails as described by @vasquesbc 

Via other tests I did get some additional information. It seems Apple have significantly changed the rules for installer packages that would run via the startosinstall tool. They now have to be both distribution installers and be signed installers. Whilst the Jamf installer is a distribution package as standard the createuser pkg is not. (This change supposedly dates back to High Sierra.) It could be the method AutoDMG is using has similar requirements.

It should be noted that with Mojave there was no issue with the equivalent installer package for creating a user.

I did get one of our in house developers to convert and sign as needed both installers and this did not seem to help. That is both startosinstall and AutoDMG still complained.

I fully appreciate Apple's desire to improve security but I do feel they have gone too far with Catalina and broken too many things. As usual they also fail to provide any documentation for administrators on how to do things properly. Frankly administering Macs is now getting to be more painful than Windows and Linux which says a lot. :( If DEP was the only approved method then they should not have methods like startosinstall or createinstallmedia. As they do have these methods they should not make them practically unusable. Apple also keep forgetting DEP is not only not available in many countries but it is impossible to add older pre-DEP purchased Macs to your account. This is wholly symptomatic with Apple's incorrect belief that -

- All users use personal laptops
- All users use iCloud
- All users use personal AppleIDs
- All enterprise users use Active Directory

PS. The -volume option for startosinstall is a new casualty in Catalina. This means startosinstall can only install on the boot drive. This is clearly an artificial and unnecessary limitation since the GUI does not have this limitation, nor does the RecoveryHD installer.
